ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -12565,15 +12566,17 @@ void analyzeInternal(ASTNode ast, Supplier<PlannerContext> pcf) throws SemanticE
     sinkOp = genOPTree(ast, plannerCtx);
 
     boolean usesMasking = false;
```

Comment:
Can you add a few masking tests that match / do not match based on exact SQL text?

There are positive and negative tests for current rewriting (masking_mv in both cases):
https://github.com/apache/hive/commit/873d31f33a061cd38be7de91b208987871fb612e#diff-3ae6a10ec619425775eeff0c135ace87cfe2c8dfbaea22f99278d7cd5d9bc330

Reply:
added tests